corehq/form_processor/models.py

```diff
@@ -892,7 +892,8 @@ def __unicode__(self):
 
     class Meta:
         index_together = [
-            ["domain", "owner_id", "closed"],
+            ["owner_id", "closed"],  # pghero recommended just indexing owner_id, but that's probably because there's few closed cases
+            ["owner_id", "server_modified_on"],
```

Comment:
Oh, awesome, right. I misinterpreted that (probably because I'm so used to couch which never _only_ ever filters with indexes), and your explanation makes total sense.